test/unit_tests/util/ObjectIOStreamTest.cpp

```diff
@@ -14,78 +14,69 @@
 #include "util/serializing/ObjectOutputStream.h"
 #include "util/serializing/Serializable.h"
 
-template <typename T, unsigned N>
-std::string serializeData(const std::array<T, N>& data) {
-    ObjectOutputStream outStream(new BinObjectEncoding);
-    outStream.writeData(&data[0], N, sizeof(T));
-    auto outStr = outStream.getStr();
-    return {outStr->str, outStr->len};
-}
-
 template <typename T>
 std::string serializeDataVector(const std::vector<T>& data) {
     ObjectOutputStream outStream(new BinObjectEncoding);
     outStream.writeData(data);
     auto outStr = outStream.getStr();
-    return {outStr->str, outStr->len};
+    auto resStr = std::string{outStr->str, outStr->len};
+    g_string_free(outStr, true);
```

Comment:
Nit: we could also wrap the g_string into a RAII guard.

Reply:
I wondered as well, but this is only in the test unit, and the real refactor would be to get rid of GString's in the ObjectOutputStream. Too big for the release branch though.